mirgecom/simutil.py

```diff
@@ -1129,6 +1137,70 @@ def configurate(config_key, config_object=None, default_value=None):
     return default_value
 
 
+def get_reasonable_array_context_class(lazy: bool = False, distributed: bool = True,
+                              profiling: bool = False):
+    if lazy and profiling:
+        raise ValueError("Can't specify both lazy and profiling")
+
+    if profiling:
+        from mirgecom.profiling import PyOpenCLProfilingArrayContext
+        return PyOpenCLProfilingArrayContext
+
+    from grudge.array_context import (get_reasonable_actx_class as
+                                      grudge_get_reasonable_actx_class)
+
+    return grudge_get_reasonable_actx_class(lazy=lazy, distributed=distributed)
+
+
+def actx_class_is_lazy(actx_class) -> bool:
+    from arraycontext import PytatoPyOpenCLArrayContext
+    return issubclass(actx_class, PytatoPyOpenCLArrayContext)
+
+
+def actx_class_is_eager(actx_class) -> bool:
+    from arraycontext import PyOpenCLArrayContext
+    return issubclass(actx_class, PyOpenCLArrayContext)
+
+
+def actx_class_is_profiling(actx_class) -> bool:
+    from mirgecom.profiling import PyOpenCLProfilingArrayContext
+    return issubclass(actx_class, PyOpenCLProfilingArrayContext)
+
+
+def initialize_actx(actx_class: Type[ArrayContext], comm: Optional["Comm"]) -> Tuple[ArrayContext, cl.Context, cl.CommandQueue, cl.tools.AllocatorBase]:
```

Comment:
Maybe only return `actx`, so it's compatible with non-CL array contexts (numpy?)? It looks like `context`, `queue`, and `allocator` are all attributes of `actx` for pyopencl/pytato, so presumably they could be retrieved in the driver that way if needed.

Reply:
Thanks, I think that makes sense. I changed it in [6578353](https://github.com/illinois-ceesd/mirgecom/pull/918/commits/65783536c02c3a01bb3d184474d11c48b85e3e87)